### DISCUSSION TITLE
Why doesn't DaisyUI read class names from JS template strings?

### DISCUSSION THREAD

**Discussion by d-rocham** [saadeghi/daisyui#1257] [Q&A] [answered]
Hi! I'm using DaisyUI together with React & Typescript. I created an `Alert.tsx` component that wraps the Alert component, and receives as one of the arguments the type of the alert (this being `info`, `error` and so forth). The `className` attribute is a template string, and the `type` prop builds the appropriate modifier class name, as shown below.

However, when rendering the application, the alert component is shown as if the class modifier isn't being read. It's only when I change the `alert-${type}` to an explicit `alert-error` that the appropriate style is shown. I already verified that the `type` prop contains the correct string.

What could cause this issue? 

```javascript
import React, { ReactElement } from 'react';

type Props = {
    message: string;
    type: string;
    icon: ReactElement;
};
function Alert({ message, type, icon }: Props) {
    return (
        <div
            className={`alert alert-${type} shadow-lg py-1 px-2 rounded-md flex flex-row w-fit mx-4`}
        >
            <div className="grow-0">{icon}</div>
            <span className="font-display">{message}</span>
        </div>
    );
}

export default Alert;
```

**Comment by saadeghi** [comment, marked as answer]
When using Tailwind, make sure your class names are not half-string/half-variable. Use classname string as a whole.  

❌ This doesn't work:
```html
<div class="alert-{{ error ? 'warning' : 'info' }}"></div>
```
✅ This works:
```html
<div class="{{ error ? 'alert-warning' : 'alert-info' }}"></div>
```
Read more here: https://tailwindcss.com/docs/content-configuration#dynamic-class-names

**Comment by daveteu** [comment]
If you really need to use the class name dynamically, u can create a object.

```
const alert = {
   warn: 'alert-warn',
   success: 'alert-success',
   error: 'alert-error',
   info: 'alert-info'
}
```

Then you can do

```
 <div  className={`alert ${alert[type]} shadow-lg py-1 px-2 rounded-md flex flex-row w-fit mx-4`}></div>
```
